src/actions/ui.js

```diff
@@ -137,10 +138,21 @@ export function closeConditionalPanel() {
   };
 }
 
-export function setProjectDirectoryRoot(url: Object) {
-  return {
-    type: "SET_PROJECT_DIRECTORY_ROOT",
-    url
+export function setProjectDirectoryRoot(newRoot: string) {
+  return ({ dispatch, getState }: ThunkArgs) => {
+    const curRoot = getProjectDirectoryRoot(getState());
+    if (newRoot.length === 0) {
```

Comment:
hmm, it would be nice to call a separate action `clearProjectDirectoryRoot` or pass in `null`